indico/modules/rb_new/client/js/common/bookings/actions.js

```diff
@@ -27,16 +29,41 @@ export const FETCH_BOOKING_DETAILS_REQUEST = 'bookings/FETCH_BOOKING_DETAILS_REQ
 export const FETCH_BOOKING_DETAILS_SUCCESS = 'bookings/FETCH_BOOKING_DETAILS_SUCCESS';
 export const FETCH_BOOKING_DETAILS_ERROR = 'bookings/FETCH_BOOKING_DETAILS_ERROR';
 
+export const BOOKING_STATE_CHANGE_REQUEST = 'bookings/BOOKING_STATE_CHANGE_REQUEST';
+export const BOOKING_STATE_CHANGE_SUCCESS = 'bookings/BOOKING_STATE_CHANGE_SUCCESS';
+export const BOOKING_STATE_CHANGE_ERROR = 'bookings/BOOKING_STATE_CHANGE_ERROR';
+export const BOOKING_STATE_UPDATED = 'bookings/BOOKING_STATE_UPDATED';
+
+export const DELETE_BOOKING_REQUEST = 'bookings/DELETE_BOOKING_REQUEST';
+export const DELETE_BOOKING_SUCCESS = 'bookings/DELETE_BOOKING_SUCCESS';
+export const DELETE_BOOKING_ERROR = 'bookings/DELETE_BOOKING_ERROR';
+
 
 export function fetchBookingDetails(id) {
-    return async (dispatch) => {
-        return await ajaxAction(
-            () => indicoAxios.get(fetchBookingDetailsURL({booking_id: id})),
-            FETCH_BOOKING_DETAILS_REQUEST,
-            [BOOKING_DETAILS_RECEIVED, FETCH_BOOKING_DETAILS_SUCCESS],
-            FETCH_BOOKING_DETAILS_ERROR,
-        )(dispatch);
-    };
+    return ajaxAction(
+        () => indicoAxios.get(fetchBookingDetailsURL({booking_id: id})),
+        FETCH_BOOKING_DETAILS_REQUEST,
+        [BOOKING_DETAILS_RECEIVED, FETCH_BOOKING_DETAILS_SUCCESS],
+        FETCH_BOOKING_DETAILS_ERROR,
+    );
 }
 
 export const openBookingDetails = (bookingId) => modalActions.openModal('booking-details', bookingId);
+
+export function changeBookingState(id, action, params = {}) {
+    return ajaxAction(
+        () => indicoAxios.post(bookingStateActionsURL({booking_id: id, action}), params),
+        BOOKING_STATE_CHANGE_REQUEST,
+        [BOOKING_STATE_CHANGE_SUCCESS, BOOKING_STATE_UPDATED],
```

Comment:
usually we trigger the "updated" action before the "success" one

Reply:
why? what is the reasoning behind that? I would say it is more correct to send the action stating that  the request was successful before any other action.

Reply:
When stepping through with the redux devtools for example, the "loading" state only goes away once the new data is available.
Also, in case we had something rendered based on the fact that a request was successful, this would be guaranteed to happen only once the new data is available (since we usually do not use the `_SUCCESS` event to update the store).

FWIW, if we had some kind of batched actions this proably wouldn't matter. But last time I checked, they had some drawbacks when it came to middlewares (since depending on the batching implemention they'd see only the batch action but not the actual underlying actions)